cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
+        self.timers = self.current_collection.find_one() 
         self.current_guild_name = ""
         self.current_date = str(date.today())
+        self.members_online = set()
+        self.timed_task = None
+    
+    @commands.Cog.listener()
+    async def on_ready(self):
+        for guild in self.bot.guilds:
+            for member in guild.members:
+                if member.status == discord.Status.online:
+                    self.members_online.add(member)
+
+        if len(self.members_online) != 0:
+            self.timed_task = asyncio.create_task(self.timed_update()) #start timed_update once bot goes online if a member is online
+
+
         
     # The 'mem_before' parameter represents the member's previous state, and 'mem_after' represents the current state.
     @commands.Cog.listener()
     async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
+        self.current_guild_name = mem_before.guild.name
+        
         if mem_before.activity and mem_before.activity == mem_after.activity:
             return
-    
-        if self.current_guild_name != mem_before.guild.name:
-            if mem_before.guild.name not in db.list_collection_names():
-                newCollection = db[mem_before.guild.name]  # If the collection doesn't exist, create a new one for the guild/server
-                self.current_guild_name = mem_before.guild.name
-                self.current_collection = newCollection               
-                newCollection.insert_one({}) # Create a new blank document in the new Collection
-                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
-            else:
-                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
-                self.current_guild_name = mem_before.guild.name
-                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+
+        if mem_before.status is discord.Status.offline and mem_after.status is discord.Status.online:
+            self.members_online.add(mem_after)
+            if self.timed_task.done():
+                self.timed_task = asyncio.create_task(self.timed_update())
+            print("added:", mem_after.name)
+        
+        elif mem_before.status is discord.Status.online and mem_after.status is discord.Status.offline:
+            self.members_online.remove(mem_before)
+            print("removed:", mem_before.name)
+            if len(self.members_online) <= 0:
+                self.timed_task.cancel()
+                try:
+                    await self.timed_task
+                except asyncio.CancelledError:
+                    print("task is now cancelled")
         
         # Track activity updates for members
         if not mem_before.activity and mem_after.activity: # activity starts/ongoing
```

Comment:
somewhere in this file, can we get a comment that has a sample dictionary layout? something like what @stephenlee91 did with his voice channel groups like ` # voice_channel_groups would look like this: { VC_Name1 : { Activity 1 : [], Activity 2 : [] } , VC_Name2 : { Activity3 : [], Activity 1 : [] } }`